zypp/sat/detail/PoolImpl.cc

```diff
@@ -505,8 +505,20 @@ namespace zypp
 	  // Add removed locales+fallback except they are still in current
 	  for ( Locale lang: localesTracker.removed() )
 	  {
-	    for ( ; lang && ! localeIds.current().count( IdString(lang) ); lang = lang.fallback() )
-	    { localeIds.removed().insert( IdString(lang) ); }
+	    if ( lang && ! localeIds.current().count( IdString(lang) ) ) {
+
+	      //remove the requested one
+	      localeIds.removed().insert( IdString(lang) );
+
+	      //remove fallbacks
+	      const auto &currLangs = localeIds.current();
+	      for ( lang = lang.fallback(); lang && ! localeIds.current().count( IdString(lang) ); lang = lang.fallback() ) {
+		//remove the language only if its not a fallback for any other
+		if ( std::none_of(currLangs.begin(), currLangs.end(), [&lang](const IdString &loc){ return Locale(loc).fallback() == lang; }) )
```

Comment:
Fallbacks form a list `de_DE->de->en`. We don't want to remove `lang` if it's in the fallbackllist of a selected locale, not only if it's the direct fallback. IMO the code would remove `en` if currLangs is just `de_DE`.

I'd enhance the `Locale` class and add a member `bool hasFallback( const Locale & fallback_r ) const;` returning whether fallback_r is in the Locales fallback list (separate commit) and use this.

Reply:
done